R/gg-plots.R

```diff
@@ -1292,6 +1292,108 @@ ggally_ratio <- function(
 
 
 
+#' Plots the number of observations
+#'
+#' Plot the number of observations by using rectangles
+#' with proportional areas.
+#'
+#' @param data data set using
+#' @param mapping aesthetics being used
+#' @param ... other arguments passed to \code{\link[ggplot2]{geom_tile}(...)}
+#' @author Joseph Larmarange \email{joseph@@larmarange.net}
+#' @keywords hplot
+#' @export
+#' @examples
+#' data(tips, package = "reshape")
+#' ggally_count(tips, mapping = ggplot2::aes(x = smoker, y = sex))
+#'
+#' ggally_count(
+#'   as.data.frame(Titanic),
+#'   mapping = ggplot2::aes(x = Class, y = Survived, weight = Freq)
+#' )
+ggally_count <- function(data, mapping, ...) {
+
+  mapping <- mapping_color_to_fill(mapping)
+  vars <- unlist(lapply(mapping, mapping_string))
+  x_var <- vars["x"]
+  y_var <- vars["y"]
+  if (is.na(x_var)) stop("'x' aesthetic is required.")
+  if (is.na(y_var)) stop("'y' aesthetic is required.")
+
+  if (is.na(vars["weight"])) {
+    data$.weight <- 1
+  } else {
+    data$.weight <- data[[vars["weight"]]]
```

Comment:
In locations where data is retrieved, please use 

https://github.com/ggobi/ggally/blob/a0b3075b41d2fe25e2f620234a5c061d9c1be8a4/R/gg-plots.R#L22-L24

ex:

```r
data$.weight <- eval_data_col(data, mapping$weight)
```

Reply:
thanks for the trick

Reply:
fix in https://github.com/ggobi/ggally/pull/321/commits/b9c71817857e713ffc4a8edf0de8ed2f35201860